simulators/active_surface.py

```diff
@@ -91,53 +133,244 @@ def get_status(self):
+
     def set_absolute_position(self, position):
         if self.delayed_execution is True:
             self.position_queue.put(self.reference_position + position)
             self.ready = True
         else:
-            self.move_to(self.reference_position + position)
+            self._move_to(self.reference_position + position)
 
     def set_relative_position(self, position):
         if self.delayed_execution is True:
             self.position_queue.put(self.current_position + position)
             self.ready = True
         else:
-            self.move_to(self.current_position + position)
+            self._move_to(self.current_position + position)
 
-    def set_io_pins(self, param):
+    def rotate(self, sign):
+        if self.running:
+            # Failed command, return False so the parser can return a byte_nak
+            return False
+        else:
+            # The following 3 lines should run in a separate thread
+            self._accel_ramp(sign)
+            self._move(sign * self.max_frequency)
+            self._decel_ramp(sign)
+            return True
+
+    def set_velocity(self, velocity):
+        # Start rotating with given velocity without an acceleration ramp
+        self.velocity = velocity
+
+        if velocity != 0:
+            self._move(velocity)
+
+    def set_stop_io(self, param):
         binary_string = bin(param)[2:].zfill(8)
 
-        self.io_dir[0] = int(binary_string[3], base=2)
-        if self.io_dir[0] == 1:
-            self.io_val[0] = int(binary_string[7], base=2)
+        # binary_string[0 - 1] is currently unused
 
-        self.io_dir[1] = int(binary_string[2], base=2)
-        if self.io_dir[1] == 1:
-            self.io_val[1] = int(binary_string[6], base=2)
+        self.stop_io_enable[0] = int(binary_string[7], 2)
+        if self.stop_io_enable[0] == 1:
+            self.stop_io_level[0] = int(binary_string[4], 2)
+        else:
+            self.stop_io_level[0] = 0
 
-        self.io_dir[2] = int(binary_string[1], base=2)
-        if self.io_dir[2] == 1:
-            self.io_val[2] = int(binary_string[5], base=2)
+        self.stop_io_enable[1] = int(binary_string[6], 2)
+        if self.stop_io_enable[1] == 1:
+            self.stop_io_level[1] = int(binary_string[3], 2)
+        else:
+            self.stop_io_level[1] = 0
 
-    def set_resolution(self, param):
-        if int(param[0]) == 1:
-            self.auto_resolution = True
-            self.resolution = self.resolutions.get(0)
+        self.stop_io_enable[2] = int(binary_string[5], 2)
+        if self.stop_io_enable[2] == 1:
+            self.stop_io_level[2] = int(binary_string[2], 2)
         else:
-            self.auto_resolution = False
-            self.resolution = self.resolutions.get(int(param[-3:], base=2))
+            self.stop_io_level[2] = 0
+
+    def set_positioning_io(self, param):
+        binary_string = bin(param)[2:].zfill(8)
+
+        # binary_string[0 - 1] is currently unused
```

Comment:
``binari_string[-1]``?  I do not get the meaning of ``[0 - 1]``.  Do you mean that the first two items are currently unused?  In that case it should be ``binary_string[0:2] is currentrly unused``.